### PR TITLE
Ali / New real account signup flow

### PR DESCRIPTION
**Pull request by ali-fs**

### CODE REVIEW

File: src/javascript/app/common/form_validation.js [hunk 1]
```diff
@@ -171,21 +173,21 @@ const Validation = (() => {
...
-    const validRegular      = (value, options) => options.regex.test(value);
-    const validEmailToken   = value => value.trim().length === 8;
-    const compareToEmail    = value => {
+    const validGeneral = value => !/[`~!@#$%^&*)(_=+[}{\]\\/";:?><|]+/.test(value);
+    const validAddress = value => !/[`~!$%^&*_=+[}{\]\\"?><|]+/.test(value);
+    const validPostCode = value => value === '' || /^[A-Za-z0-9][A-Za-z0-9\s-]*$/.test(value);
+    const validPhone = value => /^\+?((-|\s)*[0-9])*$/.test(value);
+    const validRegular = (value, options) => options.regex.test(value);
+    const validEmailToken = value => value.trim().length === 8;
+    const compareToEmail = value => {
```
Comment (reza-fs) [inline src/javascript/app/common/form_validation.js:182]: (code-style) (alignment) (throughout)

File: src/templates/app/_includes/new_account_steps/currency_form.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            <div className='align-self-center border-bottom-light-gray' />
+        </div>
+        <div className='gr-8 gr-10-p gr-12-m gr-centered invisible'>
+            <div className='gr-row gr-row-align-center gr-padding-20 gr-parent currency_list' id={`${id}_currency_list`} />
+        </div>
+    </React.Fragment>
+);
+
+const CurrencyForm = () => (
+    <React.Fragment>
```
Comment (reza-fs) [inline src/templates/app/_includes/new_account_steps/currency_form.jsx:17]: No fragment is needed here.

File: src/templates/app/_includes/new_account_steps/financial_info_form.jsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    </Fieldset>
+);
+
+const FinancialInformationForm = () => (
+    <Fieldset id='financial_information_section' legend={it.L('Financial information')}>
+        <FormRow id='income_source' label={it.L('Source of income')} type='select'>
+            <option value=''>{it.L('Please select')}</option>
+            <option value='Salaried Employee'>{it.L('Salaried Employee')}</option>
+            <option value='Self-Employed'>{it.L('Self-Employed')}</option>
+            <option value='Investments &amp; Dividends'>{it.L('Investments &amp; Dividends')}</option>
```
Comment (reza-fs) [inline src/templates/app/_includes/new_account_steps/financial_info_form.jsx:60]:
```suggestion
            <option value='Investments &amp; Dividends'>{it.L('Investments & Dividends')}</option>
```

File: src/templates/app/_includes/new_account_steps/terms_of_use_form.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+                <Jurisdiction row_id='jurisdiction_row' className='invisible' />
+                <RiskDisclaimer row_id='risk_disclaimer_row' className='invisible' />
+                <PepDeclaration row_id='pep_declaration_row' className='invisible' />
+                <Tnc row_id='tnc_row' className='invisible' />
+            </div>
+            <div className='align-end' id='submit_section'>
+                <a className='button button-secondary btn_cancel action_previous' href='javascript:;'><span>{it.L('Previous')}</span></a>
+                <button id='new_account_submit' className='button' type='submit' action='new_account'>{it.L('Add account')}</button>
+            </div>
+        </form >
```
Comment (reza-fs) [inline src/templates/app/_includes/new_account_steps/terms_of_use_form.jsx:22]: (nitpick) 
```suggestion
        </form>
```

File: src/templates/app/_includes/new_account_steps/terms_of_use_form.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import PepDeclaration from '../pep_declaration.jsx';
+import {
+    Jurisdiction,
+    RiskDisclaimer,
+    Tnc,
+} from '../../../_common/components/forms_common_rows.jsx';
+
+const TermsOfUseForm = () => (
+    <React.Fragment>
```
Comment (reza-fs) [inline src/templates/app/_includes/new_account_steps/terms_of_use_form.jsx:10]: Fragment not needed.

**Review by reza-fs — APPROVED** [review]

**Review by kevinw-binary — APPROVED** [review]

**Pull request merged**